app/code/Magento/InventoryShipping/Model/ShippingAlgorithmResultInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryShipping\Model;
+
+/**
+ * Responsible for how we will deduct product qty from different Sources
+ */
+interface ShippingAlgorithmResultInterface
+{
+    /**
+     * Returns product SKU -> source selection mapping in the following format:
```

Comment:
Introduce `SourceSelectionInterface` with `getSourceCode` method
Source code should be part of object but not key of array

Reply:
Generally, we will have next structure

``ShippingAlgorithmInterface` provides
`ShippingAlgorithmResultInterface` contains list of
`SourceSelectionInterface` (with method `getSourceCode` and `getSourceItemSelections`) contains list if
`SourceItemSelectionInterface` (without `getSourceCode`)